### ISSUE TITLE
toHaveStyleRule in jest-emotion doesn't normalize whitespace

### ISSUE THREAD

**Issue by mxmul** [emotion-js/emotion#1988]
**Current behavior:**

Emotion is stripping some whitespace from style rules, which makes it difficult to assert that a JS variable exists in emitted styles.

For example, if I have defined color token like this:
```jsx
const RED = 'rgba(255, 0, 0, 1)';
```

and I use it in a styled component, the Jest matcher considers the color value to be different:
```jsx
test('renders with correct styles', () => {
  const Div = styled('div')`
    color: ${RED};
  `;
  const tree = renderer.create(<Div />).toJSON();
  expect(tree).toHaveStyleRule('color', RED); // this fails
})
```

**To reproduce:**

1. Clone https://github.com/mxmul/jest-emotion-repo
2. `$ yarn`
3. `$ yarn test`

```jsx
$ jest
 FAIL  ./emotion.test.js
  ✕ renders with correct styles (18 ms)

  ● renders with correct styles

    Expected color to match:
      rgba(255, 0, 0, 1)
    Received:
      rgba(255,0,0,1)

      13 |   `;
      14 |   const tree = renderer.create(<Div />).toJSON();
    > 15 |   expect(tree).toHaveStyleRule('color', RED);
         |                ^
      16 | })
      17 |

      at Object.<anonymous> (emotion.test.js:15:16)
```

**Expected behavior:**

Whatever transformations Emotion is applying to CSS property values should also be applied to the values passed to `toHaveStyleRule`. It should treat `rgb(255, 0, 0)` and `rgb(255,0,0)` as equivalent.

**Environment information:**

- `react` version: 16.13.1
- `emotion` version: 10.0.35

**Comment by Andarist** [comment]
It would be great to fix this but I must say that this is not the top priority for us and I might not find the time to implement a fix for this myself.

**Comment by RoystonS** [comment]
Would a pure Stylis solution of `serialize(compile(input), [stringify])` produce the desired effect? i.e. get stylis to parse the CSS, then spit it out again using its `stringify` middleware, which does the whitespace stripping?

**Comment by Andarist** [comment]
That was literally my first intuition as well but Stylis v4 doesnt normalize whitespace in parenthesis :s

**Comment by RoystonS** [comment]
Yikes, so it'll need to be that _plus_ a little custom plugin that tidies up values inside `decl` StyleElements...

**Comment by Andarist** [comment]
Yeah, probably something like that - we could actually use a minifier plugin that would be used to optimize whitespace in our precomputed styles when using Babel plugin, so this could serve both cases.

**Comment by KitsonBroadhurst** [comment]
Just to add, I am having a similar issue, but it appears to be wider than just the whitespace.

I have a darkGreen colour in my theme,`rgb(9,73,29)` which during the test assertion is converted as follows:
![image](https://user-images.githubusercontent.com/8405069/104714821-962afd80-571d-11eb-9d61-2207b7f73a38.png)

Component css:
```
background-color: rgb('9,73,29');
```
Test assertion:
```
expect(flag).toHaveStyleRule('background', 'rgb('9,73,29')')
```

Where the hex equivalent `#136538` of the rgb value is inaccurate, I would expect `rgb(9,73,29)` to be equivalent to `#09491d`.

What's more, when I hard code that hex value (`#09491d`), I get the same conversion:
![image](https://user-images.githubusercontent.com/8405069/104715169-13567280-571e-11eb-9f3d-61c7b3ad14aa.png)

Component css:
```
background-color: #09491d;
```
Test assertion:
```
expect(flag).toHaveStyleRule('background', '#09491d')
```

**Comment by Andarist** [comment]
Please always try to share a repro case in a runnable form - either by providing a git repository to clone or a codesandbox. OSS maintainers usually can't afford the time to set up the repro, even if exact steps are given.

**Comment by KitsonBroadhurst** [comment]
> Please always try to share a repro case in a runnable form - either by providing a git repository to clone or a codesandbox. OSS maintainers usually can't afford the time to set up the repro, even if exact steps are given.

I will investigate further and reply with a repo instead!
